cmd/axelard/cmd/vald/broadcast/backlog.go

```diff
@@ -0,0 +1,71 @@
+package broadcast
+
+import (
+	"context"
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type backlog struct {
+	tail chan broadcastTask
+	head broadcastTask
+}
+
+func (bl *backlog) Pop() broadcastTask {
+	bl.loadHead()
+
+	next := bl.head
+	bl.head = broadcastTask{}
```

Comment:
because it is possible to push tasks that have no messages it is easier to deal with a dummy struct instead of nil. This way the "empty case" is the same for both.